File: backend/TodoApi/Models/TodoState.cs
namespace TodoApi.Models;

public class TodoState
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string DisplayName { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool IsDefault { get; set; } = false;
    public string? Color { get; set; }
    public string? Icon { get; set; }
    public int OrganizationId { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedAt { get; set; }
    public int? DeletedById { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Organization Organization { get; set; } = null!;
    public User? DeletedBy { get; set; }
    public ICollection<Task> Tasks { get; set; } = new List<Task>();
}

File: frontend/src/types/todoState.ts
export interface TodoState {
  id: number;
  name: string;
  displayName: string;
  order: number;
  isDefault: boolean;
  color?: string;
  icon?: string;
  organizationId: number;
  taskCount: number;
}

export interface CreateTodoStateDto {
  name: string;
  displayName: string;
  order: number;
  isDefault: boolean;
  color?: string;
  icon?: string;
}

export interface UpdateTodoStateDto {
  name?: string;
  displayName?: string;
  order?: number;
  isDefault?: boolean;
  color?: string;
  icon?: string;
}

